File: Java/BOJ/3986 좋은 단어.java
import java.io.*;
import java.util.*;
import java.math.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int result = 0;
        for (int i = 0; i < N; i++) {
        	String S = br.readLine();
        	Stack<Character> stack = new Stack<>();
        	for (int j = 0 ; j < S.length(); j++) {
        		if(stack.isEmpty()){
        			stack.push(S.charAt(j));
        		}
        		else if (stack.peek() == S.charAt(j)) {
        			stack.pop();
        		}
        		else {
        			stack.add(S.charAt(j));
        		}
        	}
        	if (stack.isEmpty()) {
        		result ++;
        	}
        }
        System.out.println(result);
    }
}

File: Python/BOJ/3986 좋은 단어.py
import sys
N = int(sys.stdin.readline())
result = 0
for _ in range(N):
    S = sys.stdin.readline().strip()
    stack = []
    for i in range(len(S)):
        if not stack:
            stack.append(S[i])
        elif stack[-1] == S[i] :
            stack.pop()
        else:
            stack.append(S[i])
    if not stack:
        result += 1
print(result)